mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -412,163 +402,183 @@ struct BlockwiseGemmV2RewritePattern
 
     int64_t KPerThread = IsKReduction ? K / inputSpansPerMfmaIn : K;
 
+    auto outerLoopM = b.create<AffineForOp>(loc, 0, mRepeats);
+    auto olmb = ConversionPatternRewriter::atBlockBegin(outerLoopM.getBody(),
+                                                        b.getListener());
+    auto olmiv = outerLoopM.getInductionVar();
+    auto mOffset = olmb.create<AddIOp>(
+        loc, aBase, olmb.create<MulIOp>(loc, mPerMfmaGroupConstantOp, olmiv));
```

Comment:
I don't think this will work for reductions as well (and since it is outside the `if`, I think you meant for it to work with reductions). I guess to unify this, we should use the same trick you mentioned in #941 , i.e., using `mfmaInsnAttr.mfmaNonKDim * mfmaInsnAttr.blocksMfma`

Reply:
I'm not sure I get your question. I believe this `mOffset` is only used in the non-reduction branch below.